src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2276,8 +2276,24 @@ public static SSTableReader moveAndOpenSSTable(ColumnFamilyStore cfs, Descriptor
             throw new RuntimeException(msg);
         }
 
-        logger.info("Renaming new SSTable {} to {}", oldDescriptor, newDescriptor);
-        SSTableWriter.rename(oldDescriptor, newDescriptor, components);
+        if (copyData)
+        {
+            try
+            {
+                logger.info("Hardlinking new SSTable {} to {}", oldDescriptor, newDescriptor);
+                SSTableWriter.hardlink(oldDescriptor, newDescriptor, components);
+            }
+            catch (Throwable t)
```

Comment:
we should not catch throwable here without rethrowing - we might hide OOM errors for example

Reply:
@krummas but that is the point, because in that catch block I am copying SSTables instead of hardlinking them. If I rethrow, how will I fall back to copy then?

Reply:
catch `FSWriteError` or what is actually thrown by `SSTableWriter.hardlink`?